### ISSUE TITLE
Make new pricing tiers reflected in feature access

### ISSUE THREAD

**Issue by dadlerj** [sourcegraph/sourcegraph#7927]
From @juliasourceress :

>Just had a client call and they're interested in our middle Enterprise Plus tier. Chatted with Dan and it sounds like we don't currently gate nor get pings back on usage of any of those "Plus" features (I think most notable ones are respecting repo permissions & extensions whitelist, but dan please chime in). Could we slot either a gate or at least a ping back/stats on those into the next release? Getting this info from clients will definitely impact new ARR as well as upsell potential.

And my response:

>Looks like we DO have enforcement for this (see https://sourcegraph.com/github.com/sourcegraph/sourcegraph@837b3b25d94e180864c7d1b7276a235a1ed9933a/-/blob/enterprise/cmd/frontend/internal/licensing/features.go#L19:2) but that we are not yet restricting it from basic Enterprise tier: https://sourcegraph.com/github.com/sourcegraph/sourcegraph@837b3b25d94e180864c7d1b7276a235a1ed9933a/-/blob/enterprise/cmd/frontend/internal/licensing/features.go#L45:23
>
>So this might be an easy update actually!
>
>Basically we just need to add the new tiers (Enterprise Plus and Elite) to our backend and update features to map to them: https://sourcegraph.com/github.com/sourcegraph/sourcegraph@837b3b2/-/blob/enterprise/cmd/frontend/internal/dotcom/billing/plans.go#L27

**Comment by unknwon** [comment]
Thanks for filing the issue! 

Have few questions:

1. Are "Enterprise Plus" and "Enterprise Elite" two different tiers?
2. Which features do we want to guard by which tiers?
3. I saw we already have "acls" for code host authorization, are we planning to treat RFC 40 "explicit repo permissions" differently?

**Comment by dadlerj** [comment]
>Are "Enterprise Plus" and "Enterprise Elite" two different tiers?

Heh, yes, you can always find all pricing info on our public site: https://about.sourcegraph.com/pricing/

or in the about repo: https://sourcegraph.com/github.com/sourcegraph/about@master/-/blob/website/src/pages/pricing.tsx

>Which features do we want to guard by which tiers?

Check out the pages above. Most important are:

1. Repo access permissions should only be available at the Enterprise Plus level
1. Extension whitelisting should only be available at the Enterprise Plus level
1. Private extension registry should only be available at the Elite level
1. Automation should only be available at the Elite level

Beyond those important ones, it'd also be nice if custom branding (the `branding` site configuration option) was only available at the Enterprise Plus level.

I don't think any other features are critical right now. @juliasourceress can you confirm?

**Comment by juliasourceress** [comment]
Thanks - confirmed.

Julia Gilinets
VP Strategic Accounts
Sourcegraph
<https://t.sidekickopen79.com/s1t/c/5/f18dQhb0S7lM8dDMPbW2n0x6l2B9nMJN7t5XWPfhMynW4XX_K63LjyBWW56dF8S3PCRs0102?te=W3R5hFj4cm2zwW4cNcV-3F6hTlW3zfP7n3Fbt5S0&si=8000000001377603&pi=617d31fb-4d13-4b24-bcee-cfbf7f897520>
| @srcgraph
<https://t.sidekickopen79.com/s1t/c/5/f18dQhb0S7lM8dDMPbW2n0x6l2B9nMJN7t5XWPfhMynW4XX_K63LjyBWW56dF8S3PCRs0102?te=W3R5hFj4cm2zwW4fQ47l4fGCmnW3Fbt5S4cPrLkf49HSj404&si=8000000001377603&pi=617d31fb-4d13-4b24-bcee-cfbf7f897520>

**Comment by tsenart** [comment]
@unknwon: Assigning this to you.

**Comment by tsenart** [comment]
Please add an estimate.

**Comment by unknwon** [comment]
I changed estimate to 1d because I discovered many features aren't actually guarded by our license tags.

**Comment by unknwon** [comment]
I'm taking this issue out of 3.13 until it's unblocked on the business side.

**Comment by juliasourceress** [comment]
sourcegraph_logo.png (https://sourcegraph.com/?utm_source=hs_email&utm_medium=email&utm_content=84707992&_hsenc=p2ANqtz-9Wb1HHV_abJ6au894Tv83RLjHeQDQLN2JsXjgfL4a2AkiwbW1zPo9yQdQPs5dXZlbvlVkPFFxe1TYv_q_4siiO4ovOBOv3vbY92gyMfJ9jeBt7aG9RWe_JbascLSObUQouaApw3N9021YteHcobQR7jP6-NQ&_hsmi=84707992 )

Hi sourcegraph/sourcegraph,

I'm Quinn (@sqs (https://twitter.com/sqs?utm_source=hs_email&utm_medium=email&utm_content=84707992&_hsenc=p2ANqtz-9Wb1HHV_abJ6au894Tv83RLjHeQDQLN2JsXjgfL4a2AkiwbW1zPo9yQdQPs5dXZlbvlVkPFFxe1TYv_q_4siiO4ovOBOv3vbY92gyMfJ9jeBt7aG9RWe_JbascLSObUQouaApw3N9021YteHcobQR7jP6-NQ&_hsmi=84707992 ) ), the Sourcegraph CEO/cofounder. We're on a mission to bring Universal Code Search (https://about.sourcegraph.com/universal-code-search?utm_source=hs_email&utm_medium=email&utm_content=84707992&_hsenc=p2ANqtz-9Wb1HHV_abJ6au894Tv83RLjHeQDQLN2JsXjgfL4a2AkiwbW1zPo9yQdQPs5dXZlbvlVkPFFxe1TYv_q_4siiO4ovOBOv3vbY92gyMfJ9jeBt7aG9RWe_JbascLSObUQouaApw3N9021YteHcobQR7jP6-NQ&_hsmi=84707992 ) to every company to help you explore, navigate, and better understand all your code. Besides working on the next release (which ships on the 20th of each month), we've been busy.

- We raised $23M (https://medium.com/craft-ventures/why-we-invested-in-sourcegraph-5ace28317e3d?utm_source=hs_email&utm_medium=email&utm_content=84707992&_hsenc=p2ANqtz-9Wb1HHV_abJ6au894Tv83RLjHeQDQLN2JsXjgfL4a2AkiwbW1zPo9yQdQPs5dXZlbvlVkPFFxe1TYv_q_4siiO4ovOBOv3vbY92gyMfJ9jeBt7aG9RWe_JbascLSObUQouaApw3N9021YteHcobQR7jP6-NQ&_hsmi=84707992 ) to bring Universal Code Search to more developers.
- Welcome to the era of Big Code (https://thenewstack.io/universal-code-search-a-new-search-tech-for-the-era-of-big-code/?utm_source=hs_email&utm_medium=email&utm_content=84707992&_hsenc=p2ANqtz-9Wb1HHV_abJ6au894Tv83RLjHeQDQLN2JsXjgfL4a2AkiwbW1zPo9yQdQPs5dXZlbvlVkPFFxe1TYv_q_4siiO4ovOBOv3vbY92gyMfJ9jeBt7aG9RWe_JbascLSObUQouaApw3N9021YteHcobQR7jP6-NQ&_hsmi=84707992 ) .
- We've grown our incredible customer community (https://about.sourcegraph.com/?utm_source=hs_email&utm_medium=email&utm_content=84707992&_hsenc=p2ANqtz-9Wb1HHV_abJ6au894Tv83RLjHeQDQLN2JsXjgfL4a2AkiwbW1zPo9yQdQPs5dXZlbvlVkPFFxe1TYv_q_4siiO4ovOBOv3vbY92gyMfJ9jeBt7aG9RWe_JbascLSObUQouaApw3N9021YteHcobQR7jP6-NQ&_hsmi=84707992#customer ) , including Uber, Lyft, F5, Qualtrics, SoFi, and Cloudflare.
- ICYMI: Yelp's engineering blog post about why Sourcegraph is vital to them (https://engineeringblog.yelp.com/2019/11/winning-the-hackathon-with-sourcegraph.html?utm_source=hs_email&utm_medium=email&utm_content=84707992&_hsenc=p2ANqtz-9Wb1HHV_abJ6au894Tv83RLjHeQDQLN2JsXjgfL4a2AkiwbW1zPo9yQdQPs5dXZlbvlVkPFFxe1TYv_q_4siiO4ovOBOv3vbY92gyMfJ9jeBt7aG9RWe_JbascLSObUQouaApw3N9021YteHcobQR7jP6-NQ&_hsmi=84707992 ) (with an awesome up-and-to-the-right chart of their weekly usage of Sourcegraph)
- Our team is now all-remote (https://about.sourcegraph.com/company/remote?utm_source=hs_email&utm_medium=email&utm_content=84707992&_hsenc=p2ANqtz-9Wb1HHV_abJ6au894Tv83RLjHeQDQLN2JsXjgfL4a2AkiwbW1zPo9yQdQPs5dXZlbvlVkPFFxe1TYv_q_4siiO4ovOBOv3vbY92gyMfJ9jeBt7aG9RWe_JbascLSObUQouaApw3N9021YteHcobQR7jP6-NQ&_hsmi=84707992 ) , with ~35 people and growing quickly (we're hiring! (https://about.sourcegraph.com/jobs/?utm_source=hs_email&utm_medium=email&utm_content=84707992&_hsenc=p2ANqtz-9Wb1HHV_abJ6au894Tv83RLjHeQDQLN2JsXjgfL4a2AkiwbW1zPo9yQdQPs5dXZlbvlVkPFFxe1TYv_q_4siiO4ovOBOv3vbY92gyMfJ9jeBt7aG9RWe_JbascLSObUQouaApw3N9021YteHcobQR7jP6-NQ&_hsmi=84707992 ) ). (We decided this before the coronavirus outbreak. Happy to help any other companies that are transitioning into working remote; just reach out to me personally.)
- The press gets it, too: see news coverage (https://about.sourcegraph.com/news?utm_source=hs_email&utm_medium=email&utm_content=84707992&_hsenc=p2ANqtz-9Wb1HHV_abJ6au894Tv83RLjHeQDQLN2JsXjgfL4a2AkiwbW1zPo9yQdQPs5dXZlbvlVkPFFxe1TYv_q_4siiO4ovOBOv3vbY92gyMfJ9jeBt7aG9RWe_JbascLSObUQouaApw3N9021YteHcobQR7jP6-NQ&_hsmi=84707992 ) .

We'll be live-streaming some of the most popular workshops that we've run with hundreds of devs at companies using Sourcegraph in the last month. Join us:

- Office hours (https://info.sourcegraph.com/office-hours-march-17?utm_source=hs_email&utm_medium=email&utm_content=84707992&_hsenc=p2ANqtz-9Wb1HHV_abJ6au894Tv83RLjHeQDQLN2JsXjgfL4a2AkiwbW1zPo9yQdQPs5dXZlbvlVkPFFxe1TYv_q_4siiO4ovOBOv3vbY92gyMfJ9jeBt7aG9RWe_JbascLSObUQouaApw3N9021YteHcobQR7jP6-NQ&_hsmi=84707992 ) - We're holding our first open forum for any topics/questions you have on Sourcegraph and Universal Code Search.

Tues Mar 17 at 10am PDT (1pm EDT)

Sign up (https://info.sourcegraph.com/office-hours-march-17?utm_source=hs_email&utm_medium=email&utm_content=84707992&_hsenc=p2ANqtz-9Wb1HHV_abJ6au894Tv83RLjHeQDQLN2JsXjgfL4a2AkiwbW1zPo9yQdQPs5dXZlbvlVkPFFxe1TYv_q_4siiO4ovOBOv3vbY92gyMfJ9jeBt7aG9RWe_JbascLSObUQouaApw3N9021YteHcobQR7jP6-NQ&_hsmi=84707992 )

- Universal Code Search (https://info.sourcegraph.com/webinar-sourcegraph-universal-code-search?utm_source=hs_email&utm_medium=email&utm_content=84707992&_hsenc=p2ANqtz-9Wb1HHV_abJ6au894Tv83RLjHeQDQLN2JsXjgfL4a2AkiwbW1zPo9yQdQPs5dXZlbvlVkPFFxe1TYv_q_4siiO4ovOBOv3vbY92gyMfJ9jeBt7aG9RWe_JbascLSObUQouaApw3N9021YteHcobQR7jP6-NQ&_hsmi=84707992 ) - Set up self-hosted Universal Code Search in <7min.

Thu Mar 19 at 10am PDT (1pm EDT)

Register now (https://info.sourcegraph.com/webinar-sourcegraph-universal-code-search?utm_source=hs_email&utm_medium=email&utm_content=84707992&_hsenc=p2ANqtz-9Wb1HHV_abJ6au894Tv83RLjHeQDQLN2JsXjgfL4a2AkiwbW1zPo9yQdQPs5dXZlbvlVkPFFxe1TYv_q_4siiO4ovOBOv3vbY92gyMfJ9jeBt7aG9RWe_JbascLSObUQouaApw3N9021YteHcobQR7jP6-NQ&_hsmi=84707992 )

- Become a Universal Code Search power user (https://info.sourcegraph.com/webinar-universal-code-search-power-user-apr-02?utm_source=hs_email&utm_medium=email&utm_content=84707992&_hsenc=p2ANqtz-9Wb1HHV_abJ6au894Tv83RLjHeQDQLN2JsXjgfL4a2AkiwbW1zPo9yQdQPs5dXZlbvlVkPFFxe1TYv_q_4siiO4ovOBOv3vbY92gyMfJ9jeBt7aG9RWe_JbascLSObUQouaApw3N9021YteHcobQR7jP6-NQ&_hsmi=84707992 ) - Many developers use code search 5x/day. If you don't, see what you're missing.

Thu Apr 2 at 10am PDT (1pm EDT)

Register now (https://info.sourcegraph.com/webinar-universal-code-search-power-user-apr-02?utm_source=hs_email&utm_medium=email&utm_content=84707992&_hsenc=p2ANqtz-9Wb1HHV_abJ6au894Tv83RLjHeQDQLN2JsXjgfL4a2AkiwbW1zPo9yQdQPs5dXZlbvlVkPFFxe1TYv_q_4siiO4ovOBOv3vbY92gyMfJ9jeBt7aG9RWe_JbascLSObUQouaApw3N9021YteHcobQR7jP6-NQ&_hsmi=84707992 )

Just want to try Sourcegraph to search your own code ASAP? Run a self-hosted Sourcegraph instance now (https://docs.sourcegraph.com/?utm_source=hs_email&utm_medium=email&utm_content=84707992&_hsenc=p2ANqtz-9Wb1HHV_abJ6au894Tv83RLjHeQDQLN2JsXjgfL4a2AkiwbW1zPo9yQdQPs5dXZlbvlVkPFFxe1TYv_q_4siiO4ovOBOv3vbY92gyMfJ9jeBt7aG9RWe_JbascLSObUQouaApw3N9021YteHcobQR7jP6-NQ&_hsmi=84707992#quickstart )  with 1 command on your own machine.

Best,

Quinn Slack

CEO, Sourcegraph

@sqs (https://twitter.com/sqs?utm_source=hs_email&utm_medium=email&utm_content=84707992&_hsenc=p2ANqtz-9Wb1HHV_abJ6au894Tv83RLjHeQDQLN2JsXjgfL4a2AkiwbW1zPo9yQdQPs5dXZlbvlVkPFFxe1TYv_q_4siiO4ovOBOv3vbY92gyMfJ9jeBt7aG9RWe_JbascLSObUQouaApw3N9021YteHcobQR7jP6-NQ&_hsmi=84707992 ) - Schedule a Zoom chat with me (https://calendly.com/sqs?utm_source=hs_email&utm_medium=email&utm_content=84707992&_hsenc=p2ANqtz-9Wb1HHV_abJ6au894Tv83RLjHeQDQLN2JsXjgfL4a2AkiwbW1zPo9yQdQPs5dXZlbvlVkPFFxe1TYv_q_4siiO4ovOBOv3vbY92gyMfJ9jeBt7aG9RWe_JbascLSObUQouaApw3N9021YteHcobQR7jP6-NQ&_hsmi=84707992 )

Sourcegraph   981 Mission St    San Francisco  CA   94103   USA

You received this email because you are subscribed to Developer Information from Sourcegraph.

Update your email preferences (https://hs-2762526.s.hubspotemail.net/hs/manage-preferences/unsubscribe?d=Vnzr2K7yMty0W2p76LK3-1rCSW3_ZXQ14pyNdBW4kFNYk4pK0K5W3Y3pvW47QCJDW49T0gg3KbsvFW3R0grW1ZszCPW3yVRfB45RlhdW3P28X24hCPvhW43T2P-6yNv9MVmWcPv90QQYPN5mdzqn25sYrW85ymKp79YB-HW8TC5qT58zcpVV1yf0N30hYWgW6YpQ9r6VCYwdW1P318D8TYYQK2133&v=3&_hsenc=p2ANqtz-9Wb1HHV_abJ6au894Tv83RLjHeQDQLN2JsXjgfL4a2AkiwbW1zPo9yQdQPs5dXZlbvlVkPFFxe1TYv_q_4siiO4ovOBOv3vbY92gyMfJ9jeBt7aG9RWe_JbascLSObUQouaApw3N9021YteHcobQR7jP6-NQ&_hsmi=84707992 ) to choose the types of emails you receive.

Unsubscribe from all future emails (https://hs-2762526.s.hubspotemail.net/hs/manage-preferences/unsubscribe-all?d=Vnzr2K7yMty0W2p76LK3-1rCSW3_ZXQ14pyNdBW4kFNYk4pK0K5W3Y3pvW47QCJDW49T0gg3KbsvFW3R0grW1ZszCPW3yVRfB45RlhdW3P28X24hCPvhW43T2P-6yNv9MVmWcPv90QQYPN5mdzqn25sYrW85ymKp79YB-HW8TC5qT58zcpVV1yf0N30hYWgW6YpQ9r6VCYwdW1P318D8TYYQK2133&v=3&_hsenc=p2ANqtz-9Wb1HHV_abJ6au894Tv83RLjHeQDQLN2JsXjgfL4a2AkiwbW1zPo9yQdQPs5dXZlbvlVkPFFxe1TYv_q_4siiO4ovOBOv3vbY92gyMfJ9jeBt7aG9RWe_JbascLSObUQouaApw3N9021YteHcobQR7jP6-NQ&_hsmi=84707992 )

**Comment by unknwon** [comment]
Hmm, what is going on😒

**Comment by unknwon** [comment]
Closing this in favor of https://github.com/sourcegraph/sourcegraph/issues/11577.
